crates/oxc_regular_expression/src/display.rs

```diff
@@ -0,0 +1,368 @@
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.body)
+    }
+}
+
+impl<'a> Display for Disjunction<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write_join(f, "|", &self.body)
+    }
+}
+
+impl<'a> Display for Alternative<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write_join(f, "", &self.body)
+    }
+}
+
+impl<'a> Display for Term<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::BoundaryAssertion(it) => write!(f, "{it}"),
+            Self::LookAroundAssertion(it) => write!(f, "{}", it.as_ref()),
+            Self::Quantifier(it) => write!(f, "{}", it.as_ref()),
+            Self::Character(it) => write!(f, "{it}"),
+            Self::Dot(it) => write!(f, "{it}"),
+            Self::CharacterClassEscape(it) => write!(f, "{it}"),
+            Self::UnicodePropertyEscape(it) => write!(f, "{}", it.as_ref()),
+            Self::CharacterClass(it) => write!(f, "{}", it.as_ref()),
+            Self::CapturingGroup(it) => write!(f, "{}", it.as_ref()),
+            Self::IgnoreGroup(it) => write!(f, "{}", it.as_ref()),
+            Self::IndexedReference(it) => write!(f, "{it}"),
+            Self::NamedReference(it) => write!(f, "{}", it.as_ref()),
+        }
+    }
+}
+
+impl Display for BoundaryAssertion {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.kind)
+    }
+}
+
+impl Display for BoundaryAssertionKind {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Start => write!(f, "^"),
+            Self::End => write!(f, "$"),
+            Self::Boundary => write!(f, "\\b"),
+            Self::NegativeBoundary => write!(f, "\\B"),
+        }
+    }
+}
+
+impl<'a> Display for LookAroundAssertion<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "({}{})", self.kind, self.body)
+    }
+}
+
+impl Display for LookAroundAssertionKind {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Lookahead => write!(f, "?="),
+            Self::NegativeLookahead => write!(f, "?!"),
+            Self::Lookbehind => write!(f, "?<="),
+            Self::NegativeLookbehind => write!(f, "?<!"),
+        }
+    }
+}
+
+impl<'a> Display for Quantifier<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.body)?;
+        match (self.min, self.max) {
+            (0, None) => write!(f, "*"),
+            (1, None) => write!(f, "+"),
+            (0, Some(1)) => write!(f, "?"),
+            (min, Some(max)) if min == max => {
+                write!(f, "{{{min}}}",)
+            }
+            (min, max) => {
+                let max = max.map_or_else(String::default, |it| it.to_string());
+                write!(f, "{{{min},{max}}}",)
+            }
+        }
```

Comment:
`self.greedy` is missing?

Reply:
Yes sir; I forgot it😅 I'll take care of this along with other changes you've suggested.